lib/encrypted-attr.js

```diff
@@ -1,13 +1,22 @@
 'use strict'
 
-const alg = 'aes-256-gcm'
-const crypto = require('crypto')
-const { get, set } = require('lodash')
+var alg = 'aes-256-gcm'
+var crypto = require('crypto')
+var get = require('lodash').get
+var set = require('lodash').set
 
 function EncryptedAttributes (attributes, options) {
   options = options || {}
 
-  let prefix = Buffer.from(`${alg}$`).toString('base64')
+  var RX_NODE_MAJOR_DOT_MINOR = /^v([^.]+\.?[^.]+)\./
+  var parsedNodeMajorAndMinorVersion = process.version.match(RX_NODE_MAJOR_DOT_MINOR) && (+(process.version.match(RX_NODE_MAJOR_DOT_MINOR)[1]))
+  var MIN_NODE_VERSION = 4
+  var isNativeCryptoFullyCapable = parsedNodeMajorAndMinorVersion >= MIN_NODE_VERSION
+  if (!isNativeCryptoFullyCapable) {
```

Comment:
So i believe `crypto.createCipheriv('aes-256-gcm',..)` will throw if the algorithm isn't supported, which is the only thing we care about. That seems like a better (and much neater) way to check.

Reply:
Also while you are here, do you mind replacing the few instances of `"aes-256-gcm"` that i missed with `alg`?